dotcom-rendering/src/model/window-guardian.ts

```diff
@@ -36,77 +36,60 @@ export interface WindowGuardianConfig {
+ * window.guardian property
+ */
+export const makeWindowGuardian = ({
+	stage,
+	frontendAssetsFullURL,
+	revisionNumber,
+	sentryPublicApiKey,
+	sentryHost,
+	keywordIds,
+	dfpAccountId,
+	adUnit,
+	ajaxUrl,
+	shouldHideReaderRevenue,
+	isPaidContent,
+	googletagUrl,
+	switches,
+	abTests,
+	editionId,
+	contentType,
+	brazeApiKey,
+	GAData,
+	unknownConfig = {},
+}: {
 	stage: StageType;
 	frontendAssetsFullURL: string;
-	page: {
-		dcrCouldRender: boolean;
-		contentType: string;
-		edition: EditionId;
-		revisionNumber: string;
-		dcrSentryDsn: string;
-		sentryHost: string;
-		sentryPublicApiKey: string;
-		keywordIds: string;
-		dfpAccountId: string;
-		adUnit: string;
-		showRelatedContent: boolean;
-		ajaxUrl: string;
-		shouldHideReaderRevenue: boolean;
-	};
-	libs: {
-		googletag: string;
-	};
+	revisionNumber: string;
+	sentryPublicApiKey: string;
+	sentryHost: string;
+	keywordIds: string;
+	dfpAccountId: string;
+	adUnit: string;
+	ajaxUrl: string;
+	googletagUrl: string;
 	switches: Switches;
-	tests: ServerSideTests;
-	ophan: {
-		pageViewId: string;
-		browserId: string;
-	};
-}
-
-const makeWindowGuardianConfig = (
-	CAPIArticle: CAPIArticleType,
-): WindowGuardianConfig => {
-	const { config } = CAPIArticle;
-	return {
-		// This indicates to the client side code that we are running a dotcom-rendering rendered page.
-		isDotcomRendering: true,
-		isDev: process.env.NODE_ENV !== 'production',
-		stage: config.stage,
-		frontendAssetsFullURL: config.frontendAssetsFullURL,
-		page: Object.assign(config, {
-			dcrCouldRender: true,
-			contentType: CAPIArticle.contentType,
-			edition: CAPIArticle.editionId,
-			revisionNumber: config.revisionNumber,
-			dcrSentryDsn:
-				'https://1937ab71c8804b2b8438178dfdd6468f@sentry.io/1377847',
-			sentryPublicApiKey: config.sentryPublicApiKey,
-			sentryHost: config.sentryHost,
-			keywordIds: config.keywordIds,
-			dfpAccountId: config.dfpAccountId,
-			adUnit: config.adUnit,
-			showRelatedContent: true,
-			ajaxUrl: config.ajaxUrl,
-		}),
-		libs: {
-			googletag: config.googletagUrl,
-		},
-		switches: config.switches,
-		tests: config.abTests,
-		ophan: {
-			pageViewId: '',
-			browserId: '',
-		},
-	} as WindowGuardianConfig;
-};
-
-export const makeWindowGuardian = (
-	CAPIArticle: CAPIArticleType,
-): {
+	abTests: ServerSideTests;
+	editionId: EditionId;
+	shouldHideReaderRevenue?: boolean;
+	isPaidContent?: boolean;
+	contentType?: string;
+	brazeApiKey?: string;
+	GAData?: GADataType;
+	/**
+	 * In the case of articles we don't know the exact values that need to exist
+	 * on the window.guardian.config.page property so rather than filter them we
+	 * allow the entire object to be passed through here and we then extend it
+	 * using Object.assigns
+	 *
+	 * This is obviously rubbish but has to be weighed against the risk of the
+	 * commercial code failing because it depended on a property we removed
+	 */
+	unknownConfig?: ConfigType | object;
```

Comment:
With the current [commercial bundle TypeScript conversion sitting above of 80%](https://guardian.github.io/commercial-bundle-progress/), this risk is much reduced than when this decision was originally made.

We should probably revisit this and see if we can move forward with an implementation. One approach might be for the @guardian/commercial-dev team to define the relevant configuration in [@guardian/commercial-core](https://github.com/guardian/commercial-core) so types can be shared accross frontend and DCR.

Reply:
Are we perhaps at the point where we could take the leap and agree a model for `config`? This would be a _huge_ win and I think it's worth some time investigating